cmd/genbindings/clang2il.go

```diff
@@ -512,8 +553,16 @@ func processEnum(node map[string]interface{}, addNamePrefix string) (CppEnum, er
 	// Underlying type
 	ret.UnderlyingType = parseSingleTypeString("int")
 	if nodefut, ok := node["fixedUnderlyingType"].(map[string]interface{}); ok {
-		if nodequal, ok := nodefut["qualType"].(string); ok {
-			ret.UnderlyingType = parseSingleTypeString(nodequal)
+		var desugared, qualType string
+		if d, ok := nodefut["desugaredQualType"].(string); ok {
+			desugared = d
+		}
+		if q, ok := nodefut["qualType"].(string); ok {
+			qualType = q
+		}
+		qualType = getPreferredType(desugared, qualType)
```

Comment:
I think this function could take the node `interface{}` as a parameter, that would simplify all the call sites.

Reply:
Agreed. Unreasonably sloppy on my part. This should be ready for review now.